trin-core/src/cli.rs

```diff
@@ -187,17 +194,38 @@ impl TrinConfig {
             val => panic!("Unsupported json-rpc protocol: {}", val),
         }
 
-        match config.geth_url {
+        match config.pandaops_url {
             Some(_) => {
-                if config.trusted_provider != TrustedProviderType::Geth {
-                    panic!("--geth-url flag is incompatible without Geth as the trusted provider.")
+                if config.trusted_provider != TrustedProviderType::Pandaops {
+                    panic!("--pandaops-url flag is incompatible without pandaops as the trusted provider.")
                 }
             }
             None => {
-                if config.trusted_provider == TrustedProviderType::Geth {
-                    panic!("Using Geth for a trusted provider requires a --geth-url flag.")
+                if config.trusted_provider == TrustedProviderType::Pandaops {
+                    panic!("Using pandaops for a trusted provider requires a --pandaops-url flag.")
+                }
+            }
+        }
+        if config.local_provider_url.is_some()
+            && config.trusted_provider != TrustedProviderType::Local
+        {
+            panic!("--local-provider-url flag is incompatible without 'local' as the trusted provider.")
+        }
+        if config.trusted_provider == TrustedProviderType::Local
+            && config.local_provider_url.is_none()
+        {
+            panic!("'--trusted-provider local' choice requires the --local-provider-url flag.")
+        }
+        match &config.local_provider_url {
```

Comment:
I think we can take advantage of the [url](https://docs.rs/url/2.3.1/url/index.html) library here (which `ureq` uses). This will help eliminate some tricky edge cases with matching schemes, domains, ports, etc.

Reply:
Good call

Reply:
Done